File: 프로그래머스/src/프로그래머스_12212021/로또의최고순위와최저순위.java
package 프로그래머스_12212021;
/** @author June Park
 *  알고리즘 스터디 - 12/21/2021
 *  프로그래머스 LV 1 문제
 * */


import java.util.Arrays;


public class 로또의최고순위와최저순위 {

static int[] rank = {6,6,5,4,3,2,1}; //rank[맞은개수] = 순위 rank[1개맞음]=6 rank[6]=1위
	
	public static void main(String[] args) {
		int[] lottos = {44, 1, 0, 0, 31, 25};
		int[] win_nums = {31, 10, 45, 1, 6, 19};
		solution(lottos,win_nums);
	}
	// 출력 형식 : 최저등수 최고 등수
	public static void solution(int[] lottos, int[] win_nums) {
	
		
		int possibilities = 0;
		for(int i = 0 ; i<lottos.length ; i++) {
			if(lottos[i]==0) {possibilities++;} 
		}//0의 개수를 구함
		
		int same = 0;
		//일치 여부 확인
		for(int i = 0 ; i<lottos.length;i++) {
			if(lottos[i]<=0)continue;
			for(int j = 0  ; j<win_nums.length; j++) {
				if (win_nums[j]<0) continue;
				else if(lottos[i]==win_nums[j]) {
					same ++;
					lottos[i]=win_nums[j]=-1;
					break;
				}
			}
		}
	//	System.out.printf(" same : %d poss+same : %d",same, (same+possibilities));
	//	System.out.printf("\nrank: %d highest %d", rank[same],rank[same+possibilities]);
		int[] answer = { rank[same+possibilities],rank[same]};
		System.out.println(Arrays.toString(answer));
	}

}

File: 프로그래머스/src/프로그래머스_02242022/로또의_최고순위와_최저순위.cpp
#include <string>
#include <vector>

using namespace std;
/*lottos : 낙서된 번호. 
  win_nums : 우승 번호
  알아볼 수 없는 번호  : 0, 번호 6개의 최고순위와 최저순위(순서무관)
  최저 순위 : 둘다 아닐때. 즉, 안지워진 번호가 최저 순위이다
  최고 순위 : 다 맞췄을 때. 즉, 전체 - 틀린개수 
 */
vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    //1등  : 6개 일치, 2등 : 5개 ... 6등 :1개 or 0개
    vector<int> rank = {6,6,5,4,3,2,1}; //rank[맞춘 개수] = 랭크
    int possible = 0;
    int match = 0;
    //step 1 : lottos의 0의 개수 파악 및 맞춘 개수 파악
    for(auto i = 0; i<6 ; i++)
    {
        if(lottos[i]==0) possible++;
        for(int j = 0 ; j<6 ; j++)
        {
            if(win_nums[j]==lottos[i])
            {
                match ++;
                break;
            }
        }
    }

    //step 2: 최저개수 및 최고개수 파악
    vector<int> answer;//최고랭크, 최저랭크
    answer.push_back(rank[match+possible]);
    answer.push_back(rank[match]);

    return answer;
}
